cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -81,7 +81,7 @@ scalar_as_column_view::return_type scalar_as_column_view::operator()<cudf::strin
                            1,
                            nullptr,
                            (bitmask_type const*)s.validity_data(),
-                           cudf::UNKNOWN_NULL_COUNT,
+                           static_cast<size_type>(s.is_valid(stream)),
```

Comment:
This should be inverted:
```suggestion
                           static_cast<size_type>(!s.is_valid(stream)),
```

Reply:
Also, please modify the live above it, using `static_cast` instead of C-style cast.

Reply:
Done.